src/test/java/com/cdancy/bitbucket/rest/features/BranchApiMockTest.java

```diff
@@ -100,6 +114,33 @@ public void testListBranches() throws Exception {
         }
     }
 
+
+    private static class BuildStatusInfo {
```

Comment:
Not that I agree with json parsing internals leakage into API model, but I've done as you suggested.
BTW https://github.com/OpenFeign/feign from netflix + some lombok seems much cleaner to build client library